docs/reference-guides/interactivity-api/core-concepts/how-the-interactivity-api-works

```diff
@@ -0,0 +1,276 @@
+# How the Interactivity API works: An In-Depth Look
+
+To get a complete picture of how the Interactivity API works, let's review a request to an imagined `/interactive` page.
+
+The `/interactive` page is a standard page that contains an accordion block among static content. In this
+case, the request will pass through the following phases:
+
+## Server-side processing begins
+
+### 1. Initial processing
+
+The initial request to the `/interactive` page is processed by WordPress, and all the hooks are fired as usual.
+
+### 2. HTML DOM processing
+
+At this stage, the current theme generates a markup for the current request. During it, the place with the interactive
+accordion block
+is reached:
+
+```php
+// render.php
+<?php wp_interactivity_state('accordion', [
+  'isOpen'             => false,
+  'isLastItemSet'      => false,
+  'lastOpenedItemName' => "",
+]); ?>
+
+<div class="accordion"
+     data-wp-interactive="accordion"
+<?php echo wp_interactivity_data_wp_context(["openItemMessage" => __('Current open item is', 'domain'), "closedItemMessage" => __('Items are closed.', 'domain'), "lastOpenItemMessage"=> __('Last opened item is', 'domain'),]); ?>
+      data-wp-class--dark="state.isDark">
+    <div class='accordion__panel'>
+        <div class='accordion__heading-opened' data-wp-bind--hidden="!state.isOpen">
+            <span data-wp-text="context.openItemMessage"></span>&nbsp;
+            <span class='accordion__open-item-name' data-wp-text="state.lastOpenedItemName"></span>
+        </div>
+        <div class='accordion__heading-closed' data-wp-bind--hidden="state.isOpen">
+            <span data-wp-text="context.closedItemMessage"></span>
+            <p class='accordion__closed-item' data-wp-bind--hidden="!state.isLastItemSet">
+                <span data-wp-text="context.lastOpenItemMessage"></span>&nbsp;
+                <span class='accordion__closed-item-name' data-wp-text="state.lastOpenedItemName"></span>
+            </p>
+        </div>
+    </div>
+
+    <div class='accordion__item' data-wp-context='{"isItemClosed":true,"itemName":"First"}'>
+        <p class='accordion__item-title' data-wp-on--click="actions.toggleItem" data-wp-text="context.itemName"></p>
+        <div class='accordion__item-content' data-wp-bind--hidden="context.isItemClosed">Content of the first item</div>
+    </div>
+
+    <div class='accordion__item' data-wp-context='{"isItemClosed":true, "itemName":"Second"}'>
+        <p class='accordion__item-title' data-wp-on--click="actions.toggleItem" data-wp-text="context.itemName"></p>
+        <div class='accordion__item-content' data-wp-bind--hidden="context.isItemClosed">Content of the second item
+        </div>
+    </div>
+</div> 
+```
+
+Note: The `wp_interactivity_state` and `wp_interactivity_data_wp_context` calls are optional,
```

Comment:
It should be made clear that `wp_interactivity_state` and `wp_interactivity_data_wp_context` are NOT optional in this example. The Accordion would not work without them. 

They are only "optional" in the sense that they are not required for the Interactivity API to work.